diesel_derives/tests/queryable_by_name.rs

```diff
@@ -32,11 +41,12 @@ fn tuple_struct() {
     struct MyStruct(
         #[diesel(column_name = foo)] i32,
         #[diesel(column_name = bar)] i32,
+        #[diesel(column_name = r#type)] i32,
     );
 
     let conn = &mut connection();
-    let data = sql_query("SELECT 1 AS foo, 2 AS bar").get_result(conn);
-    assert_eq!(Ok(MyStruct(1, 2)), data);
+    let data = sql_query("SELECT 1 AS foo, 2 AS bar, 3 AS \"r#type\"").get_result(conn);
```

Comment:
I think to be consistent with previous comment we probably should search for `type` as name if passing the `r#type` ident

and ideally also allow passing `column_name = "some arbitrary string"` where we would then parse `3 as "some arbitrary string"` even if `some arbitrary string` is not an ident to support other weird things users might want. (I'm not sure if that is already supported and it's not a blocker here)

Reply:
I have lifted the restriction on `column_name` insofar, that a raw identifier is now allowed, however it must still correspond to a valid Rust identifier.
The previous error-message for invalid names is retained, though it now emits an unhelpful
```
error: Expected valid identifier, found `ty pe`. Diesel automatically renames invalid identifiers, perhaps you meant to write `ty pe_`?
   --> diesel_derives/tests/as_changeset.rs:287:32
    |
287 |         #[diesel(column_name = "ty pe")]
    |                                ^^^^^^^
```
The error-message should probably use `inference::rust_name_for_sql_name()`?